src/k2/module/k23si/client/k23si_client.h

```diff
@@ -370,23 +407,40 @@ class K2TxnHandle {
         return _cpo_client->partitionRequest
             <dto::K23SIWriteRequest, dto::K23SIWriteResponse, dto::Verbs::K23SI_WRITE>
             (_options.deadline, *request).
-            then([this, request=std::move(request)] (auto&& response) {
+            then([this, request=std::move(request), isonehot=std::move(isonehot)] (auto&& response) {
                 auto& [status, k2response] = response;
 
                 _registerRangeForWrite(status, *request);
 
                 _checkResponseStatus(status);
                 _ongoing_ops--;
 
-                if (status.is2xxOK() && !_heartbeat_timer.isArmed()) {
+                k2::CachedSteadyClock::duration sleep= 0us; // time to sleep before we return to satisfy min time_spent in txn
+
+                if (!isonehot && status.is2xxOK() && !_heartbeat_timer.isArmed()) {
                     K2ASSERT(log::skvclient, _cpo_client->collections.find(_trh_collection) != _cpo_client->collections.end(), "collection not present after successful partial update");
                     K2LOG_D(log::skvclient, "Starting hb, mtr={}", _mtr)
                     _heartbeat_interval = _cpo_client->collections[_trh_collection]->collection.metadata.heartbeatDeadline / 2;
                     _makeHeartbeatTimer();
                     _heartbeat_timer.armPeriodic(_heartbeat_interval);
                 }
+                else if(isonehot) { //handle one hot transactions separately and similar to what is done in end()
+                    if (status.is2xxOK()) {
+                        _client->successful_txns++;
+                    }
+                    //note we need not end heartbeat timer because we never started it for one hot transactions.
+                    auto time_spent = Clock::now() - _start_time;
+                    if (time_spent < 5us) {
+                        sleep = 5us - time_spent;
+                    }
+                }
 
-                return seastar::make_ready_future<PartialUpdateResult>(PartialUpdateResult(std::move(status)));
+                return seastar::do_with( (k2::CachedSteadyClock::duration)sleep, [sleep=std::move(sleep),status=std::move(status)] (k2::CachedSteadyClock::duration& dur) {
```

Comment:
same issues as in the non-partial use case. You can probably just remove this for now since YCSB does only full-key updates, no partials right?

If not, you should try to refactor this a bit

Reply:
No YCSB does partial updates. A particular field is chosen at random and updated.